Purchases/Public/RCPurchases.h

```diff
@@ -13,15 +13,26 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+
+
+typedef void (^RCReceivePurchaserInfoBlock)(RCPurchaserInfo * _Nullable, NSError * _Nullable);
+typedef void (^RCReceiveIntroEligibilityBlock)(NSDictionary<NSString *, RCIntroEligibility *> *);
+typedef void (^RCReceiveEntitlementsBlock)(NSDictionary<NSString *,RCEntitlement *> * _Nullable, NSError * _Nullable);
+typedef void (^RCReceiveProductsBlock)(NSArray<SKProduct *> *);
+typedef void (^RCPurchaseCompletedBlock)(SKPaymentTransaction * _Nullable, RCPurchaserInfo * _Nullable, NSError * _Nullable);
```

Comment:
All these `_Nullable` and the `NSError` in all the blocks, make me think it would be a good idea to have a 

```
typedef void (^RCErrorBlock)(NSError *);
```

And make the functions accept two blocks, a success block and an error block. What do you think?

Reply:
Thinking more about this, maybe is a good idea to leave it a single block. In Android, the Kotlin compiler will create a lambda from an interface with a single function, so it will look better. On the other hand, the Nullable kinda loses the meaning, since there's no way to say either RCPurchaserInfo is null or the error is null. In the past I have used an Either class stolen from the functional world that indicates that either the left is null or the right (either purchaserinfo is null or the error, but not both at the same time). I am not saying we should use Either, but that it would be nice if we find a way to express which one is null or not (I don't think there is)